### PR TITLE
230222 feat(quic): add hidden low level settings for listeners.

### PR DESCRIPTION
**Pull request by qzhuyan**
Fixes [EMQX-9001](https://emqx.atlassian.net/browse/EMQX-9001)

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [ ] Changed lines covered in coverage report
- [x] Change log has been added to `changes/<version>/(feat|fix)-<PR-id>.en.md` and `.zh.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [ ] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [x] Schema changes are backward compatible

### CODE REVIEW

File: apps/emqx/src/emqx_schema.erl [hunk 1]
```diff
@@ -862,6 +865,81 @@ fields("mqtt_quic_listener") ->
...
+        {"handshake_idle_timeout_ms",
+            quic_lowlevel_settings_uint(
+                1,
+                ?MAX_UINT(64),
+                "How long a handshake can idle before it is discarded. Default: 10 000"
+            )},
+        {"tls_server_max_send_buffer",
+            quic_lowlevel_settings_uint(
+                1,
+                ?MAX_UINT(64),
```
Comment (qzhuyan) [inline apps/emqx/src/emqx_schema.erl:885]: uint32

File: apps/emqx/src/emqx_schema.erl [hunk 2]
```diff
@@ -862,6 +865,81 @@ fields("mqtt_quic_listener") ->
...
+                "The size (in packets) of the initial congestion window for a connection. Default: 10"
+            )},
+        {"send_idle_timeout_ms",
+            quic_lowlevel_settings_uint(
+                1,
+                ?MAX_UINT(32),
+                "Reset congestion control after being idle for amount of time. Default: 1000"
+            )},
+        {"initial_rtt_ms",
+
```
Comment (qzhuyan) [inline apps/emqx/src/emqx_schema.erl:925]: remove

File: apps/emqx/src/emqx_schema.erl [hunk 3]
```diff
@@ -862,6 +865,81 @@ fields("mqtt_quic_listener") ->
...
+            quic_lowlevel_settings_uint(
+                1,
+                ?MAX_UINT(32),
+                "Initial RTT estimate."
```
Comment (qzhuyan) [inline apps/emqx/src/emqx_schema.erl:929]: default 333

File: apps/emqx/src/emqx_schema.erl [hunk 4]
```diff
@@ -886,6 +976,105 @@ fields("mqtt_quic_listener") ->
                     desc => ?DESC(fields_mqtt_quic_listener_keep_alive_interval)
                 }
             )},
+        {"keep_alive_interval_ms",
+            quic_lowlevel_settings_uint(
+                1,
+                ?MAX_UINT(32),
+                "How often to send PING frames to keep a connection alive."
```
Comment (qzhuyan) [inline apps/emqx/src/emqx_schema.erl:983]: default 0, dsiabled.

File: apps/emqx/src/emqx_schema.erl [hunk 5]
```diff
@@ -886,6 +976,105 @@ fields("mqtt_quic_listener") ->
...
+            )},
+        {"peer_bidi_stream_count",
+            quic_lowlevel_settings_uint(
+                1,
+                ?MAX_UINT(16),
+                "Number of bidirectional streams to allow the peer to open."
+            )},
+        {"peer_unidi_stream_count",
+            quic_lowlevel_settings_uint(
+                1,
```
Comment (qzhuyan) [inline apps/emqx/src/emqx_schema.erl:993]: 0

File: apps/emqx/src/emqx_schema.erl [hunk 6]
```diff
@@ -886,6 +976,105 @@ fields("mqtt_quic_listener") ->
                     desc => ?DESC(fields_mqtt_quic_listener_keep_alive_interval)
                 }
             )},
+        {"keep_alive_interval_ms",
+            quic_lowlevel_settings_uint(
+                1,
```
Comment (qzhuyan) [inline apps/emqx/src/emqx_schema.erl:981]: 0

File: apps/emqx/src/emqx_schema.erl [hunk 7]
```diff
@@ -870,6 +948,12 @@ fields("mqtt_quic_listener") ->
                     desc => ?DESC(fields_mqtt_quic_listener_idle_timeout)
                 }
             )},
+        {"idle_timeout_ms",
+            quic_lowlevel_settings_uint(
+                1,
```
Comment (qzhuyan) [inline apps/emqx/src/emqx_schema.erl:953]: 0

File: apps/emqx/src/emqx_listeners.erl [hunk 8]
```diff
@@ -726,3 +727,57 @@ get_ssl_options(Conf) ->
...
+%% @see quicer:quic_settings()
+-spec optional_quic_listener_opts(map()) -> proplists:proplist().
+optional_quic_listener_opts(Conf) when is_map(Conf) ->
+    maps:to_list(
+        maps:filter(
+            fun(Name, _V) ->
+                lists:member(
+                    Name,
+                    [
+                        max_bytes_per_key,
```
Comment (zmstone) [inline apps/emqx/src/emqx_listeners.erl:741]: nit: move it to a function?
`quic_listener_opt_names() -> [.....]`

File: apps/emqx/src/emqx_schema.erl [hunk 9]
```diff
@@ -862,6 +865,80 @@ fields("mqtt_quic_listener") ->
                 }
             )},
         {"ciphers", ciphers_schema(quic)},
+
+        {"max_bytes_per_key",
+            quic_lowlevel_settings_uint(
+                1,
+                ?MAX_UINT(64),
+                "Maximum number of bytes to encrypt with a single 1-RTT encryption key"
```
Comment (zmstone) [inline apps/emqx/src/emqx_schema.erl:873]: should move them to `desc` with zh translations.

File: apps/emqx/i18n/emqx_schema_i18n.conf [hunk 10]
```diff
@@ -1901,6 +1901,347 @@ base_listener_acceptors {
...
+    }
+    label {
+        en: "MTU discovery missing probe count"
+        zh: "MTU发现丢失的探针数量"
+    }
+}
+
+fields_mqtt_quic_listener_max_binding_stateless_operations {
+    desc {
+        en: "The maximum number of stateless operations that may be queued ona binding at any one time. Default: 100"
```
Comment (ieQu1) [inline apps/emqx/i18n/emqx_schema_i18n.conf:2225]:
```suggestion
        en: "The maximum number of stateless operations that may be queued on a binding at any one time. Default: 100"
```
?

File: apps/emqx/i18n/emqx_schema_i18n.conf [hunk 11]
```diff
@@ -1901,6 +1901,347 @@ base_listener_acceptors {
     }
 }
 
+fields_mqtt_quic_listener_max_bytes_per_key {
+    desc {
+        en: "Maximum number of bytes to encrypt with a single 1-RTT encryption keybefore initiating key update. Default: 274877906944"
```
Comment (ieQu1) [inline apps/emqx/i18n/emqx_schema_i18n.conf:1906]:
```suggestion
        en: "Maximum number of bytes to encrypt with a single 1-RTT encryption key before initiating key update. Default: 274877906944"
```

Comment (qzhuyan) [reply apps/emqx/i18n/emqx_schema_i18n.conf:1906]: thanks, I auto generated them from last commit the newlines are forgotten. :smile:

**Review by zmstone — APPROVED** [review]

**Pull request merged**